### PR TITLE
feat: allow styles to be specified for MenuGroup labels

### LINKED ISSUE #2720: Allow styles to be specified for MenuGroup labels
### Description

Allow styles to be specified for MenuGroup labels

### Problem Statement/Justification

Currently users cannot specify their own style for the label portion, so

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by illionillion**
Closes #2720  

## Description

Allow styles to be specified for MenuGroup labels.
And update test code.

### CODE REVIEW

File: packages/components/menu/src/menu-group.tsx [hunk 1]
```diff
@@ -8,10 +8,14 @@ export type MenuGroupProps = HTMLUIProps<"div"> & {
    * The label of the group.
    */
   label?: string
+  /**
+   * The label of the group style.
```
Comment (hirotomoyamada) [inline packages/components/menu/src/menu-group.tsx:12]: Change "The label of the group style." to "Props for menu group element.".

File: packages/components/menu/src/menu-group.tsx [hunk 2]
```diff
@@ -8,10 +8,14 @@ export type MenuGroupProps = HTMLUIProps<"div"> & {
...
+   */
+  labelStyle?: HTMLUIProps<"span">
```
Comment (hirotomoyamada) [inline packages/components/menu/src/menu-group.tsx:14]: Change `labelStyle` to `labelProps`.

File: packages/components/menu/src/menu-option-group.tsx [hunk 3]
```diff
@@ -11,6 +11,10 @@ type MenuOptionGroupOptions<Y extends string | string[] = string> = {
    * The label of the option group.
    */
   label?: string
+  /**
+   * The label of the group style.
+   */
+  labelStyle?: HTMLUIProps<"span">
```
Comment (hirotomoyamada) [inline packages/components/menu/src/menu-option-group.tsx:17]: This is not necessary.Instead, merge the `MenuGroup` type definitions.

**Comment by hirotomoyamada** [comment]
@illionillion 

Hi, thanks for the PR.

Would you modify `Autocomplete` and `Select` as well? Or do you work with a different PR?

**Comment by illionillion** [comment]
> @illionillion
> 
> Hi, thanks for the PR.
> 
> Would you modify `Autocomplete` and `Select` as well? Or do you work with a different PR?

I'm thinking of creating a new PR.

If I pass group data in an array in items, how should I specify labelProps?

Should it be possible to specify in an array in addition to OptionGroup?

```tsx
const items: SelectItem[] = [
  { label: "ベジータ", value: "ベジータ" },
  {
    label: "地球人",
    labelProps: {
      // custom label  style
    },
    items: [
      { label: "孫悟空", value: "孫悟空" },
      { label: "孫悟飯", value: "孫悟飯" },
      { label: "クリリン", value: "クリリン" },
    ],
  },
  {
    label: "フリーザ軍",
    labelProps: {
      // custom label  style
    },
    items: [
      { label: "フリーザ", value: "フリーザ" },
      { label: "ギニュー", value: "ギニュー" },
      { label: "リクーム", value: "リクーム" },
      { label: "バータ", value: "バータ" },
      { label: "ジース", value: "ジース" },
      { label: "グルド", value: "グルド" },
    ],
  },
]

return <Select placeholder="キャラクターを選択" items={items} />
```

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**